packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -277,11 +305,28 @@ function FlowsheetDataGrid(props) {
       return;
     }
 
+    const gridRef = dataGridFuncRef.current.getGridRef();
+
+    let anchorSectionId = '';
+    if (flowsheetSections) {
+      for (let rowIdx = rowIndex; rowIdx > 0; rowIdx -= 1) {
+        const currentSectionId = gridRef.rows[rowIdx].getAttribute('data-section-id');
+        if (currentSectionId) {
+          anchorSectionId = currentSectionId;
+          break;
+        }
+      }
+    }
```

Comment:
Why can't the handleCellRangeSelection method provide the section id?  Wouldn't that already be available without you having to query the grid?